providers/MapService.js

```diff
@@ -0,0 +1,117 @@
+         * @property {string} searchField - The Field name that is to be searched.
+         */
+        searchField: '',
+
+        /**
+         * @property {string} contextField - The context field
+         */
+        contextField: null,
+
+        /**
+         * @property {Query} _query
+         * @private
+         */
+        _query: null,
+
+        /**
+         * @property {QueryTask} _queryTask
+         * @private
+         */
+        _queryTask: null,
+
+        /**
+         * A provider class for Sherlock that searches a layer within a mapservice.
+         * @name MapService
+         * @param {string} url - The URL to the map service layer that you want to search.
+         * e.g. http://mapserv.utah.gov/arcgis/rest/services/SomeService/MapServer/1
+         * @param {string} searchField - The name of the field that you want to search on.
+         * @param options {object}
+         * @param {string[]} [options.outFields] - A list of the fields that you want returned from the search.
+         * If undefined then the searchField and contextField (if provided) will be used.
+         * @param {string} [options.contextField] - A second field to display in the results table to
+         * give context to the results in case of duplicate results.
+         * @param {string} [options.token] - Token for working with secured services
+         */
+        constructor: function (url, searchField, options) {
+            console.log('sherlock.providers.MapService:constructor', arguments);
+
+            var that = this;
+
+            this.searchField = searchField;
+            var outFields;
+            if (options) {
+                this.contextField = options.contextField;
+                outFields = options.outFields;
+                // add token for secured services
+                if (options.token) {
+                    url += '/?token=' + this.token ;
+                }
+            }
+
+            this._query = new Query();
+            this._query.returnGeometry = false;
+            this._query.outFields = this._getOutFields(outFields, this.contextField, this.searchField)
+
+            this._queryTask = new QueryTask(url);
+            this._queryTask.on('error', function handleQueryTaskError(er) {
+                that.emit('error', er);
+            });
+        },
+        /**
+         * Initiates a search for features
+         * @param {string} searchString - The text to search for
+         * @returns {Deferred}
+         */
+        search: function (searchString) {
+            console.log('sherlock.providers.MapService:search', arguments);
+
+            this._query.returnGeometry = false;
+            this._query.where = this._getSearchClause(searchString);
+
+            this._deferred = this._queryTask.execute(this._query)
+                .then(function handleQueryTaskResponse(featureSet) {
+                    return featureSet.features;
+                });
+
+            return this._deferred;
```

Comment:
An array of graphics. I'm not sure if this is most agnostic choice but it's easy for the map service and webapi providers to return.